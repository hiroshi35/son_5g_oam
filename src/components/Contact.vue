<template>
  <div>
    <panel :header="false">
      <div class="contact">
        <div class="contact-avatar">
          <img class="img-circle center-block" :src="image" width="70px" height="70px">
        </div>
        <div class="row">
          <div class="col-md-12">
            <h3>{{name}}</h3>
            <small class="margin-bottom-10 dis-bl" v-text="title"></small>
            <div class="social-buttons">
              <a href="#" class="btn  btn-fb rounded"><i class="fa fa-facebook" aria-hidden="true"></i></a>
              <a href="#" class="btn  btn-gplus rounded"><i class="fa fa-google" aria-hidden="true"></i></a>
              <a href="#" class="btn  btn-twitter rounded"><i class="fa fa-twitter" aria-hidden="true"></i></a>
            </div>

            <div class="btn-group btn-group-outlined-margin btn-group-xs" role="group" aria-label="...">
              <button type="button" class="btn btn-outlined-default rounded-left">
                <i class="fa fa-envelope" aria-hidden="true"></i>
                Send A Message
              </button>
              <button type="button" class="btn btn-outlined-default rounded-right">View Profile <i class="fa fa-user"
                                                                                                   aria-hidden="true"></i>
              </button>
            </div>
          </div>
        </div>
      </div>
    </panel>
  </div>
</template>

<script>
  import Panel from './Panel.vue'

  export default {
    components: {
      Panel
    },
    props: {
      name: {
        type: String,
        required: true
      },
      title: {
        type: String,
        required: true
      },
      image: {
        type: String,
        required: true
      }
    },
    data () {
      return {}
    }
  }
</script>