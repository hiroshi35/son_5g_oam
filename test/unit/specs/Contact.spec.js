import { mount } from 'vue-test-utils'
import Contact from '@/components/Contact.vue'

describe('Contact.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(Contact,
      {
        propsData: {
          name: 'Adam Simpson',
          title: 'CEO',
          image: '/static/imgs/avatars/man-3.png'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('has correct props', () => {
    expect(wrapper.props().name).toBe('Adam Simpson')
    expect(wrapper.props().title).toBe('CEO')
    expect(wrapper.props().image).toBe('/static/imgs/avatars/man-3.png')
  })
})
