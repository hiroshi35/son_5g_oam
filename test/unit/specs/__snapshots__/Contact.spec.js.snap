// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact.vue has the expected html structure 1`] = `
<div>
  <div
    class="panel panel-white"
  >
    <!---->
     
    <div
      class="panel-body"
    >
      <div
        class="contact"
      >
        <div
          class="contact-avatar"
        >
          <img
            class="img-circle center-block"
            height="70px"
            src="/static/imgs/avatars/man-3.png"
            width="70px"
          />
        </div>
         
        <div
          class="row"
        >
          <div
            class="col-md-12"
          >
            <h3>
              Adam Simpson
            </h3>
             
            <small
              class="margin-bottom-10 dis-bl"
            >
              CEO
            </small>
             
            <div
              class="social-buttons"
            >
              <a
                class="btn  btn-fb rounded"
                href="#"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-facebook"
                />
              </a>
               
              <a
                class="btn  btn-gplus rounded"
                href="#"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-google"
                />
              </a>
               
              <a
                class="btn  btn-twitter rounded"
                href="#"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-twitter"
                />
              </a>
            </div>
             
            <div
              aria-label="..."
              class="btn-group btn-group-outlined-margin btn-group-xs"
              role="group"
            >
              <button
                class="btn btn-outlined-default rounded-left"
                type="button"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-envelope"
                />
                
              Send A Message
            
              </button>
               
              <button
                class="btn btn-outlined-default rounded-right"
                type="button"
              >
                View Profile 
                <i
                  aria-hidden="true"
                  class="fa fa-user"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;
